src/cliffordsynthesis/CliffordSynthesizer.cpp

```diff
@@ -395,4 +417,100 @@ void CliffordSynthesizer::updateResults(const Configuration& config,
     break;
   }
 }
+
+void gateToLayer(const qc::Operation& gate, std::size_t& i,
+                 std::vector<std::size_t>& layers,
+                 std::vector<std::size_t>& layerNum, std::size_t& layer) {
+  for (const auto& qubit : gate.getUsedQubits()) {
+    if (layerNum[qubit] >= layer) {
+      ++layer;
+      layers.emplace_back(i);
+      break;
+    }
+  }
+  for (const auto& qubit : gate.getUsedQubits()) {
+    layerNum[qubit] = layer;
+  }
+  ++i;
+}
+
+// assume canonical sorting of gates
+std::vector<std::size_t> getLayers(const qc::QuantumComputation& qc) {
+  std::vector<std::size_t> layerNum{};
+  layerNum.resize(qc.size());
```

Comment:
I would be careful here. Since it's a vector of `std::size_t`, passing a size to the constructor creates a vector with one element instead of a vector with a pre-defined size. So this is the correct way here.

Reply:
Right. Nevermind then 😌